vortex-vector/src/primitive/generic_mut_impl.rs

```diff
@@ -131,6 +133,24 @@ impl<T: NativePType> PVectorMut<T> {
             }
         }
     }
+
+    /// Returns the remaining spare capacity of the vector as a slice of [`MaybeUninit<T>`].
+    ///
+    /// The returned slice can be used to fill the buffer with data before marking the data as
+    /// initialized using unsafe methods like [`set_len`].
+    ///
+    /// # Safety
+    ///
+    /// Note that this only provides access to the spare capacity of the **elements** buffer.
+    ///
+    /// After writing to the spare capacity and calling [`set_len`], the caller must also ensure the
+    /// validity mask is updated accordingly to maintain consistency.
+    ///
+    /// [`set_len`]: Self::set_len
+    #[inline]
+    pub unsafe fn spare_capacity_mut(&mut self) -> &mut [MaybeUninit<T>] {
```

Comment:
Do we need this? MaybeUninit is a really horrible API to work with, particularly for primitives where just calling reserve + set_len + as_mut seems to be the way most things go